### ISSUE TITLE
Unable to Retrieve Patient Data from FHIR Service When Authentication is Configured for FHIR & DICOM App Services.

### ISSUE THREAD

**Issue by HitakshiDobariya99** [microsoft/dicom-server#3178]
I've deployed [DICOM Cast](https://github.com/microsoft/dicom-server/blob/main/docs/quickstarts/deploy-dicom-cast.md#deploy-via-azure-with-oss) successfully. And I have performed all the steps of [Sync Medical Imaging Server for DICOM metadata into FHIR Server for Azure](https://github.com/microsoft/dicom-server/blob/main/docs/how-to-guides/sync-dicom-metadata-to-fhir.md). 

I cannot fetch FHIR resources when I set up [authentication](https://github.com/microsoft/dicom-server/blob/main/docs/how-to-guides/sync-dicom-metadata-to-fhir.md#set-the-authentication-for-your-fhir--dicom-app-services) in the FHIR service. It is working fine when I skip that step. Do you have any ideas on why I'm unable to get FHIR resources?

**Comment by HitakshiDobariya99** [comment]
hello @poadhika 

Could you please look into this matter ASAP? Your support in this matter will be appreciated.

**Comment by poadhika** [comment]
It seems like you have another issue opened for the same matter, I am closing this issue as we are communicating in this thread. https://github.com/microsoft/dicom-server/issues/3180

**Comment by HitakshiDobariya99** [comment]
I've deployed [DICOM Cast](https://github.com/microsoft/dicom-server/blob/main/docs/quickstarts/deploy-dicom-cast.md#deploy-via-azure-with-oss) successfully. And I have performed all the steps of [Sync Medical Imaging Server for DICOM metadata into FHIR Server for Azure](https://github.com/microsoft/dicom-server/blob/main/docs/how-to-guides/sync-dicom-metadata-to-fhir.md). The configuration involves three steps:

1. Create Resource App
2. Set Authentication
3. Update Key Value

Successfully completing all three steps should retrieve the corresponding FHIR resource (Patient's data) in the FHIR service. 

However, I'm facing an issue where following all the steps doesn't yield the expected result. 
Surprisingly, if I skip the second step (set authentication), I can retrieve the FHIR resource successfully.

This issue might be interlinked with a previous matter I described, and resolving that might automatically fix this integration issue. Looking for guidance and solutions to ensure seamless integration.
